### ISSUE TITLE
Names of features and implementations mixed up

### ISSUE THREAD

**Issue by lwrage** [osate/osate-ge#129]
Take the following model
```
package test
public
	system S
		features
			i: in data port;
			o: out data port;
	end S;
	
	system implementation S.i
	end S.i;
end test;
```
and open the package diagram or the diagram for S.i. Several bad things happen because the editor seems to mix up S.i (the system implementation) and S.i (the feature i in S).

**Comment by philip-alldredge** [comment]
The serializable reference service needs to take into account the type of the referenced object with performing lookups.

**Comment by philip-alldredge** [comment]
Fixed in b5111d219bda99eea88403aa72efd101b989561e.
